Unable to use platformio for zephyr framework builds

### Describe the bug
I'm unable to build Zephyr-based projects with platformio, because it sets up its own virtual environment and installs packages using `pip`. This doesn't work on NixOS.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Install `platformio`
2. Setup Zephyr project with `pio project init --board nrf52840_dk --project-option "framework=zephyr"`
3. `pio run`

I encounter this error:

```console
$ pio run
Processing nrf52840_dk (platform: nordicnrf52; board: nrf52840_dk; framework: zephyr)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/nordicnrf52/nrf52840_dk.html
PLATFORM: Nordic nRF52 (8.1.0) > Nordic nRF52840-DK
HARDWARE: NRF52840 64MHz, 256KB RAM, 1MB Flash
DEBUG: Current (jlink) On-board (cmsis-dap, jlink) External (blackmagic, stlink)
PACKAGES:
 - framework-zephyr 2.20500.210226 (2.5.0)
 - framework-zephyr-canopennode 0.0.0-alpha+sha.468d350028
 - framework-zephyr-civetweb 0.0.0-alpha+sha.e6903b80c0
 - framework-zephyr-cmsis 0.0.0-alpha+sha.c3bd2094f9
 - framework-zephyr-fatfs 0.0.0-alpha+sha.1d1fcc725a
 - framework-zephyr-hal-nordic 0.0.0-alpha+sha.f0d54d8449
 - framework-zephyr-hal-st 0.0.0-alpha+sha.b52fdbf4b6
 - framework-zephyr-libmetal 0.0.0-alpha+sha.9d4ee2c3cf
 - framework-zephyr-littlefs 0.0.0-alpha+sha.9e4498d1c7
 - framework-zephyr-loramac-node 0.0.0-alpha+sha.3f545d76a2
 - framework-zephyr-lvgl 0.0.0-alpha+sha.31acbaa36e
 - framework-zephyr-mbedtls 0.0.0-alpha+sha.24d84ecff1
 - framework-zephyr-mcuboot 0.0.0-alpha+sha.3fc59410b6
 - framework-zephyr-mcumgr 0.0.0-alpha+sha.43845e883f
 - framework-zephyr-mipi-sys-t 0.0.0-alpha+sha.75e671550a
 - framework-zephyr-open-amp 0.0.0-alpha+sha.de1b85a130
 - framework-zephyr-openthread 0.0.0-alpha+sha.1d668284a0
 - framework-zephyr-segger 0.0.0-alpha+sha.38c79a447e
 - framework-zephyr-sof 0.0.0-alpha+sha.b5b772dd61
 - framework-zephyr-tfm-mcuboot 1.7.0-rc1
 - framework-zephyr-tinycbor 0.0.0-alpha+sha.40daca97b4
 - framework-zephyr-tinycrypt 0.0.0-alpha+sha.3e9a49d267
 - framework-zephyr-trusted-firmware-m 0.0.0-alpha+sha.96340fb6c0
 - tool-cmake 3.16.4
 - tool-dtc 1.4.7
 - tool-gperf 3.0.4
 - tool-ninja 1.7.1
 - tool-sreccat 1.164.0 (1.64)
 - toolchain-gccarmnoneeabi 1.80201.181220 (8.2.1)
Installing Zephyr's Python dependencies
/nix/store/yqjhkdarf7p74wcha7aypzsb751n7wl2-python3-3.9.5/bin/python3.9: No module named pip
*** Error 1
Reading CMake configuration...
-- Application: /home/j-hui/scratch/pio-proj/zephyr
-- Zephyr version: 2.5.0 (/home/j-hui/.platformio/packages/framework-zephyr)
-- Found Python3: /nix/store/yqjhkdarf7p74wcha7aypzsb751n7wl2-python3-3.9.5/bin/python3.9 (found suitable exact version "3.9.5") found components: Interpreter
-- Configuring incomplete, errors occurred!

CMake Error at /home/j-hui/.platformio/packages/framework-zephyr/cmake/zephyr_module.cmake:61 (message):
  Traceback (most recent call last):

    File "/home/j-hui/.platformio/packages/framework-zephyr/scripts/zephyr_module.py", line 25, in <module>
      import pykwalify.core

  ModuleNotFoundError: No module named 'pykwalify'

Call Stack (most recent call first):
  /home/j-hui/.platformio/packages/framework-zephyr/cmake/app/boilerplate.cmake:183 (include)
  CMakeLists.txt:2 (include)
```

### Expected behavior
It should install the `framework-zephyr` package, before proceeding to build the package and the project.

### Notify maintainers
@mogorman 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.52, NixOS, 21.05.1669.973910f5c31 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.14`
 - channels(root): `"nixos-21.05.1669.973910f5c31, nixpkgs-unstable-21.11pre304227.ecaf3da9340"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/by-name/pl/platformio-chrootenv/package.nix
+++ b/pkgs/by-name/pl/platformio-chrootenv/package.nix
@@ -17,11 +17,18 @@ let
       xdg-user-dirs
       ncurses
       udev
+      # Required for esp-idf and other frameworks that manage their own
+      # Python virtual environments during the build process.
+      # See: https://github.com/NixOS/nixpkgs/issues/133185
+      cmake
+      ninja
     ])
     ++ (with python.pkgs; [
       python
       setuptools
       pip
+      wheel
+      virtualenv
       bottle
     ]);
 

